dart/lib/src/client.dart

```diff
@@ -77,6 +74,16 @@ abstract class SentryClient {
     return _options.transport.send(event);
   }
 
+  SentryEvent _prepareEvent(SentryEvent event) => event.copyWith(
+        serverName: event.serverName ?? _options.serverName,
+        dist: event.dist ?? _options.dist,
+        environment:
+            event.environment ?? _options.environment ?? defaultEnvironment,
+        release: event.release ?? _options.release,
+        sdk: event.sdk ?? Sdk(name: sdkName, version: sdkVersion),
```

Comment:
I meant to read the sdk from the options:

https://github.com/getsentry/sentry-dart/pull/140#discussion_r512756637

Reply:
`Sdk(name: sdkName, version: sdkVersion)` this could be done on options.sdk field declaration or ctor body so its never null?

Reply:
with `Sdk(name: sdkName, version: sdkVersion)` as default value ?
should we do the same with defaultEnvironment and initiate options.environment with a default value ?

Reply:
yes, as default value see here: https://github.com/getsentry/sentry-java/blob/main/sentry/src/main/java/io/sentry/SentryOptions.java#L1107

no about defaultEnvironment, one can overwrite the `null` and we don't want that, unless you add a null checker on the setter, then it should be fine

Reply:
✅